### ISSUE TITLE
Can't poll LaserJet color flow MFP M575

### ISSUE THREAD

**Issue by ruben-herold** [librenms/librenms#17166]
### The problem

hi,

response: printer-schmeinta : xmt/rcv/%loss = 3/3/0%, min/avg/max = 0.61/0.878/1.4  
SNMP['/usr/bin/snmpget' '-M' '/opt/librenms/mibs' '-m' 'SNMPv2-TC:SNMPv2-MIB:IF-MIB:IP-MIB:TCP-MIB:UDP-MIB:NET-SNMP-VACM-MIB' '-v3' '-l' 'authPriv' '-n' "" '-x' 'AES' '-X' 'XXXXX' '-a' 'SHA' '-A' 'XXXX' '-u' 'admin' '-OQXUte' '-Pu' 'udp:printer-schmeinta:161' 'SNMPv2-MIB::sysObjectID.0']  
Exitcode: 1 ["snmpget: Bad context specified
"] 

this printer needs context jetdirect to work

### Output of ./validate.php

```txt
[librenms@librenms ~]$ ./validate.php 
===========================================
Component | Version
--------- | -------
LibreNMS  | 25.1.0-114-gac01305ae (2025-02-14T06:28:45+01:00)
DB Schema | 2025_01_07_223946_drop_cisco_a_s_a_table (315)
PHP       | 8.3.15
Python    | 3.9.21
Database  | MariaDB 10.5.27-MariaDB
RRDTool   | 1.7.2
SNMP      | 5.9.1
===========================================

[OK]    Composer Version: 2.8.5
[OK]    Dependencies up-to-date.
[OK]    Database connection successful
[OK]    Database connection successful
[OK]    Database Schema is current
[OK]    SQL Server meets minimum requirements
[OK]    lower_case_table_names is enabled
[OK]    MySQL engine is optimal
[OK]    Database and column collations are correct
[OK]    Database schema correct
[OK]    MySQL and PHP time match
[OK]    Active pollers found
[OK]    Dispatcher Service not detected
[OK]    Locks are functional
[OK]    Python poller wrapper is polling
[OK]    Redis is unavailable
[OK]    rrd_dir is writable
[OK]    rrdtool version ok
```

### Device info

LaserJet color flow MFP M575 

### Discovery output

```txt
./discovery.php -d -h printer-schmeinta
LibreNMS Discovery
SQL[select `migration` from `migrations` order by `id` desc limit 1 [] 0.37ms] 
  
SQL[select count(*) as aggregate from `migrations` [] 0.32ms] 
  
SQL[select version() [] 0.22ms] 
  
===========================================
Component | Version
--------- | -------
LibreNMS  | 25.1.0-114-gac01305ae (2025-02-14T06:28:45+01:00)
DB Schema | 2025_01_07_223946_drop_cisco_a_s_a_table (315)
PHP       | 8.3.15
Python    | 3.9.21
Database  | MariaDB 10.5.27-MariaDB
RRDTool   | 1.7.2
SNMP      | 5.9.1
===========================================
DEBUG!
LibreNMS config cache cleared and config reloaded.  
SQL[select `config_name`, `config_value` from `config` [] 0.41ms] 
  
SQL[select * from `graph_types` [] 0.27ms] 
  
SQL[select * from `config` where (`config_name` = ?) limit 1 ["fping6"] 0.36ms] 
  
SQL[delete from `config` where `config_name` like ? ["fping6.%"] 0.23ms] 
  
SQL[select * from `config` where (`config_name` = ?) limit 1 ["traceroute"] 0.34ms] 
  
SQL[delete from `config` where `config_name` like ? ["traceroute.%"] 0.21ms] 
  
SQL[SELECT * FROM `devices` WHERE disabled = 0 AND `hostname` LIKE 'printer-schmeinta' ORDER BY device_id DESC [] 0.58ms] 
  
printer-schmeinta 40 jetdirect SQL[select * from `devices` where `device_id` = ? limit 1 [40] 0.42ms] 
  
SQL[select * from `devices_attribs` where `devices_attribs`.`device_id` = ? and `devices_attribs`.`device_id` is not null [40] 0.29ms] 
  
[FPING] '/usr/sbin/fping' '-4' '-e' '-q' '-c' '3' '-p' '500' '-t' '500' '-O' '0' 'printer-schmeinta'
  
response: printer-schmeinta : xmt/rcv/%loss = 3/3/0%, min/avg/max = 0.658/1.4/2.78  
SNMP['/usr/bin/snmpget' '-M' '/opt/librenms/mibs' '-m' 'SNMPv2-TC:SNMPv2-MIB:IF-MIB:IP-MIB:TCP-MIB:UDP-MIB:NET-SNMP-VACM-MIB' '-v3' '-l' 'authPriv' '-n' "" '-x' 'AES' '-X' 'PASSWORD' '-a' 'SHA' '-A' 'PASSWORD' '-u' 'USER' '-OQXUte' '-Pu' 'udp:HOSTNAME:161' 'SNMPv2-MIB::sysObjectID.0']  
Exitcode: 1 ["snmpget: Bad context specified
"] 
  
snmpget: Bad context specified
  
SNMP query failed. Exit Code: 1 Empty: true Bad String: not found  
./discovery.php printer-schmeinta 2025-02-14 14:08:08 - 0 devices discovered in 1.307 secs  

SNMP [1/0.05s]: Snmpget[1/0.05s]  
SQL [13/0.05s]: Select[11/0.05s] Delete[2/0.00s]  
RRD [0/0.00s]:
```

### Poller output

```txt
SQL[select `migration` from `migrations` order by `id` desc limit 1 [] 0.46ms] 
  
SQL[select count(*) as aggregate from `migrations` [] 0.4ms] 
  
SQL[select version() [] 0.19ms] 
  
===========================================
Component | Version
--------- | -------
LibreNMS  | 25.1.0-114-gac01305ae (2025-02-14T06:28:45+01:00)
DB Schema | 2025_01_07_223946_drop_cisco_a_s_a_table (315)
PHP       | 8.3.15
Python    | 3.9.21
Database  | MariaDB 10.5.27-MariaDB
RRDTool   | 1.7.2
SNMP      | 5.9.1
===========================================
  
LibreNMS config cache cleared and config reloaded.  
SQL[select `config_name`, `config_value` from `config` [] 0.48ms] 
  
SQL[select * from `graph_types` [] 0.3ms] 
  
SQL[select * from `config` where (`config_name` = ?) limit 1 ["fping6"] 0.4ms] 
  
SQL[delete from `config` where `config_name` like ? ["fping6.%"] 0.24ms] 
  
SQL[select * from `config` where (`config_name` = ?) limit 1 ["traceroute"] 0.39ms] 
  
SQL[delete from `config` where `config_name` like ? ["traceroute.%"] 0.25ms] 
  
Starting polling run:

SQL[select `device_id` from `devices` where `hostname` = ? ["printer-schmeinta"] 0.39ms] 
  
SQL[select * from `devices` where `device_id` = ? limit 1 [40] 0.54ms] 
  
Hostname:  printer-schmeinta  (printer)
ID:        40
OS:        jetdirect
IP:        10.10.8.211
  
Attempting to initialize OS: jetdirect  
OS initialized: LibreNMS\OS\Jetdirect  
SQL[select * from `devices_attribs` where `devices_attribs`.`device_id` = ? and `devices_attribs`.`device_id` is not null [40] 1.84ms] 
  
[FPING] '/usr/sbin/fping' '-4' '-e' '-q' '-c' '3' '-p' '500' '-t' '500' '-O' '0' 'printer-schmeinta'
  
response: printer-schmeinta : xmt/rcv/%loss = 3/3/0%, min/avg/max = 0.643/1.16/2.18  
SNMP['/usr/bin/snmpget' '-M' '/opt/librenms/mibs' '-m' 'SNMPv2-TC:SNMPv2-MIB:IF-MIB:IP-MIB:TCP-MIB:UDP-MIB:NET-SNMP-VACM-MIB' '-v3' '-l' 'authPriv' '-n' "" '-x' 'AES' '-X' 'PASSWORD' '-a' 'SHA' '-A' 'PASSWORD' '-u' 'USER' '-OQXUte' '-Pu' 'udp:HOSTNAME:161' 'SNMPv2-MIB::sysObjectID.0']  
Exitcode: 1 ["snmpget: Bad context specified
"] 
  
snmpget: Bad context specified
  
SNMP query failed. Exit Code: 1 Empty: true Bad String: not found  
SQL[update `devices` set `last_ping` = ?, `last_ping_timetaken` = ? where `device_id` = ? ["2025-02-14 14:08:41",1.16,40] 1.17ms] 
  
RRD[update /opt/librenms/rrd/printer-schmeinta/icmp-perf.rrd N:1.16:3:3:0.643:2.18]  
SQL[select * from `device_outages` where `device_outages`.`device_id` = ? and `device_outages`.`device_id` is not null and `up_again` is null order by `going_down` desc limit 1 [40] 0.47ms] 
  
#### Load poller module availability ####
  
Module enabled: Global + | OS   | Device   | Manual    
SQL[select * from `device_outages` where `device_outages`.`device_id` = ? and `device_outages`.`device_id` is not null and `up_again` >= ? order by `going_down` asc [40,1739452121] 0.37ms] 
  
SQL[select * from `availability` where (`device_id` = ? and `duration` = ?) limit 1 [40,86400] 0.35ms] 
  
SQL[update `availability` set `availability_perc` = ? where `availability_id` = ? [66.825,161] 4.77ms] 
  
RRD[update /opt/librenms/rrd/printer-schmeinta/availability-86400.rrd N:66.825]  
1 day   : 66.825%  
SQL[select * from `device_outages` where `device_outages`.`device_id` = ? and `device_outages`.`device_id` is not null and `up_again` >= ? order by `going_down` asc [40,1738933721] 0.41ms] 
  
SQL[select * from `availability` where (`device_id` = ? and `duration` = ?) limit 1 [40,604800] 0.26ms] 
  
SQL[update `availability` set `availability_perc` = ? where `availability_id` = ? [66.825,162] 0.78ms] 
  
RRD[update /opt/librenms/rrd/printer-schmeinta/availability-604800.rrd N:66.825]  
1 week  : 66.825%  
SQL[select * from `device_outages` where `device_outages`.`device_id` = ? and `device_outages`.`device_id` is not null and `up_again` >= ? order by `going_down` asc [40,1736946521] 0.33ms] 
  
SQL[select * from `availability` where (`device_id` = ? and `duration` = ?) limit 1 [40,2592000] 0.3ms] 
  
SQL[update `availability` set `availability_perc` = ? where `availability_id` = ? [66.825,163] 8.64ms] 
  
RRD[update /opt/librenms/rrd/printer-schmeinta/availability-2592000.rrd N:66.825]  
4 weeks : 66.825%  
SQL[select * from `device_outages` where `device_outages`.`device_id` = ? and `device_outages`.`device_id` is not null and `up_again` >= ? order by `going_down` asc [40,1708002521] 0.39ms] 
  
SQL[select * from `availability` where (`device_id` = ? and `duration` = ?) limit 1 [40,31536000] 0.31ms] 
  
SQL[update `availability` set `availability_perc` = ? where `availability_id` = ? [66.825,164] 0.82ms] 
  
RRD[update /opt/librenms/rrd/printer-schmeinta/availability-31536000.rrd N:66.825]  
11 months : 66.825%  
SQL[delete from `availability` where `availability`.`device_id` = ? and `availability`.`device_id` is not null and `availability_id` not in (?, ?, ?, ?) [40,161,162,163,164] 0.24ms] 
  
  
>> SNMP: [1/0.05s] MySQL: [16/0.21s] RRD: [5/0.00s]  
>> Runtime for poller module 'availability': 0.0268 seconds with 150080 bytes  
RRD[update /opt/librenms/rrd/printer-schmeinta/poller-perf-availability.rrd N:0.026833057403564]  
#### Unload poller module availability ####
  
SQL[select * from `device_graphs` where `device_graphs`.`device_id` = ? and `device_graphs`.`device_id` is not null [40] 0.42ms] 
  
Enabled graphs (13): availability poller_modules_perf ping_perf netstat_snmp netstat_snmp_pkt netstat_udp netstat_tcp poller_perf uptime netstat_icmp netstat_icmp_info netstat_ip netstat_ip_frag

  

>>> Polled printer-schmeinta (40) in 1.099 seconds <<<  
#### Start Alerts ####  
SQL[select `device_groups`.*, `device_group_device`.`device_id` as `pivot_device_id`, `device_group_device`.`device_group_id` as `pivot_device_group_id` from `device_groups` inner join `device_group_device` on `device_groups`.`id` = `device_group_device`.`device_group_id` where `device_group_device`.`device_id` = ? [40] 0.47ms] 
  
SQL[select exists(select * from `alert_schedule` where (`start` <= ? and `end` >= ? and (`recurring` = ? or (`recurring` = ? and ((time(`start`) < time(`end`) and time(`start`) <= ? and time(`end`) > ?) or (time(`start`) > time(`end`) and (time(`start`) <= ? or time(`end`) > ?))) and (`recurring_day` like ? or `recurring_day` is null)))) and (exists (select * from `devices` inner join `alert_schedulables` on `devices`.`device_id` = `alert_schedulables`.`alert_schedulable_id` where `alert_schedule`.`schedule_id` = `alert_schedulables`.`schedule_id` and `alert_schedulables`.`alert_schedulable_type` = ? and `alert_schedulables`.`alert_schedulable_id` = ?))) as `exists` ["2025-02-14T13:08:41.635980Z","2025-02-14T13:08:41.635980Z",0,1,"13:08:41","13:08:41","13:08:41","13:08:41","%","device",40] 0.64ms] 
  
SQL[select * from `devices` where `devices`.`device_id` = ? limit 1 [40] 0.49ms] 
  
SQL[SELECT DISTINCT a.* FROM alert_rules a
        LEFT JOIN alert_device_map d ON a.id=d.rule_id AND (a.invert_map = 0 OR a.invert_map = 1 AND d.device_id = ?)
        LEFT JOIN alert_group_map g ON a.id=g.rule_id AND (a.invert_map = 0 OR a.invert_map = 1 AND g.group_id IN (SELECT DISTINCT device_group_id FROM device_group_device WHERE device_id = ?))
        LEFT JOIN alert_location_map l ON a.id=l.rule_id AND (a.invert_map = 0 OR a.invert_map = 1 AND l.location_id IN (SELECT DISTINCT location_id FROM devices WHERE device_id = ?))
        LEFT JOIN devices ld ON l.location_id=ld.location_id AND ld.device_id = ?
        LEFT JOIN device_group_device dg ON g.group_id=dg.device_group_id AND dg.device_id = ?
        WHERE a.disabled = 0 AND (
            (d.device_id IS NULL AND g.group_id IS NULL AND l.location_id IS NULL)
            OR (a.invert_map = 0 AND (d.device_id=? OR dg.device_id=? OR ld.device_id=?))
            OR (a.invert_map = 1  AND (d.device_id != ? OR d.device_id IS NULL) AND (dg.device_id != ? OR dg.device_id IS NULL) AND (ld.device_id != ? OR ld.device_id IS NULL))
        ) [40,40,40,40,40,40,40,40,40,40,40] 1.1ms] 
  
#### End Alerts (0.0108s) ####
  
### Start Device Groups ###  
SQL[select * from `device_groups` [] 0.28ms] 
  
SQL[select * from `device_group_device` where `device_group_device`.`device_id` = ? [40] 0.27ms] 
  
Groups Added:   Removed:   
### End Device Groups (0.0014s) ### 
  
Device was down, unable to poll.
```

**Comment by murrant** [comment]
Which snmp queries need the context of jetdirect?

**Comment by murrant** [comment]
Is this only an issue with snmp v3?

**Comment by laf** [comment]
I believe this is fixed in a recent version.
